geode-core/src/distributedTest/java/org/apache/geode/internal/cache/wan/asyncqueue/AsyncEventQueuePausedDUnitTest.java

```diff
@@ -0,0 +1,242 @@
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.core.ConditionTimeoutException;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import org.apache.geode.DataSerializable;
+import org.apache.geode.DataSerializer;
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.RegionShortcut;
+import org.apache.geode.cache.asyncqueue.internal.AsyncEventQueueImpl;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+import org.apache.geode.internal.cache.wan.MyAsyncEventListener;
+import org.apache.geode.test.dunit.rules.ClientVM;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.categories.AEQTest;
+
+@Category({AEQTest.class})
+@RunWith(Parameterized.class)
+public class AsyncEventQueuePausedDUnitTest implements Serializable {
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        // RegionType , isParallel AEQ
+        {RegionShortcut.PARTITION, true},
+        {RegionShortcut.PARTITION, false},
+        {RegionShortcut.REPLICATE, false},
+        {RegionShortcut.PARTITION_PERSISTENT, true},
+        {RegionShortcut.PARTITION_PERSISTENT, false},
+        {RegionShortcut.PARTITION_REDUNDANT, true},
+        {RegionShortcut.PARTITION_REDUNDANT, false},
+        {RegionShortcut.REPLICATE_PERSISTENT, false},
+        {RegionShortcut.PARTITION_REDUNDANT_PERSISTENT, true},
+        {RegionShortcut.PARTITION_REDUNDANT_PERSISTENT, false}
+    });
+  }
+
+  @Parameterized.Parameter
+  public static RegionShortcut regionShortcut;
+
+  @Parameterized.Parameter(1)
+  public static boolean isParallel;
+
+  @Rule
+  public ClusterStartupRule lsRule = new ClusterStartupRule(4);
+
+  private static MemberVM locator, server1, server2;
+  private static ClientVM client;
+
+  @Before
+  public void beforeClass() throws Exception {
+    locator = lsRule.startLocatorVM(0);
+    server1 = lsRule.startServerVM(1, "group1", locator.getPort());
+    server2 = lsRule.startServerVM(2, "group1", locator.getPort());
+    int serverPort = server1.getPort();
+    client =
+        lsRule.startClientVM(3, new Properties(),
+            clientCacheFactory -> configureClientCacheFactory(clientCacheFactory, serverPort));
+  }
+
+  private static void configureClientCacheFactory(ClientCacheFactory ccf, int... serverPorts) {
+    for (int serverPort : serverPorts) {
+      ccf.addPoolServer("localhost", serverPort);
+    }
+    ccf.setPoolReadTimeout(10 * 60 * 1000); // 10 min
+    ccf.setPoolSubscriptionEnabled(true);
+  }
+
```

Comment:
Can we add a test; to verify the persistent aeq, when recovered doesn't invoke listener.

Reply:
doing it now.

Reply:
I have it with the PR for the gfsh commands. will put up a PR asap